pkg/api/v1/types.go

```diff
@@ -3089,6 +3089,13 @@ type Secret struct {
 	// Described in https://tools.ietf.org/html/rfc4648#section-4
 	Data map[string][]byte `json:"data,omitempty" protobuf:"bytes,2,rep,name=data"`
 
+	// StringData allows specifying non-binary secret data in string form.
+	// It is provided as a write-only convenience method.
+	// All keys and values are moved to the data field on write.
```

Comment:
yeah, merged. stringData wins... it's write-only, so any time you specify it, you're saying you want it. that lets patch work as intended (you can patch a field in either map and it takes effect)